### ISSUE TITLE
multiprocessing async .get() hangs if the pool is terminated

### ISSUE THREAD

**Issue by mdavis-xyz** [python/cpython#126743]
# Bug report

### Bug description:

# MWE
## Steps to reproduce:

```python
import multiprocessing
from time import sleep

with multiprocessing.Pool() as p:
    result = p.apply_async(sleep, (1,))
result.get()
```

## Expected behavior:

Either:

* result is returned. (`Pool().__exit__` waits for every async task to finish); or
* some kind of exception is raised when `.get()` is called outside `with Pool()`

# Comments

This is the same as #79659. That was marked as completed in 2021. However it seems the issue is still present.

### CPython versions tested on:

3.12, 3.13, 3.14

### Operating systems tested on:

Linux

**Comment by YvesDup** [comment]
> I think we need to think about [@vstinner](https://github.com/vstinner) [#11139](https://github.com/python/cpython/pull/11139) . It's clean and useful I think

I agree, but I'm thinking of something simpler:  at the begining of the `ApplyResult.get` method, we just have to test if `pool._state` is terminated (or closed), and then raises a `RuntimeError` exception.

**Comment by YvesDup** [comment]
@Zheaoli would you want to use the [@vstinner](https://github.com/vstinner) https://github.com/python/cpython/pull/11139 old aborted proposition as a new PR ?
